src/sync.cpp

```diff
@@ -77,52 +77,28 @@ boost::thread_specific_ptr<LockStack> lockstack;
 
 static void potential_deadlock_detected(const std::pair<void*, void*>& mismatch, const LockStack& s1, const LockStack& s2)
 {
-    // We attempt to not assert on probably-not deadlocks by assuming that
-    // a try lock will immediately have otherwise bailed if it had
-    // failed to get the lock
-    // We do this by, for the locks which triggered the potential deadlock,
-    // in either lockorder, checking that the second of the two which is locked
-    // is only a TRY_LOCK, ignoring locks if they are reentrant.
-    bool firstLocked = false;
```

Comment:
Do you need to get rid of the if (!fTry) condition below in push_lock() to make this happen with trylocks?

Also, it doesn't appear to that invlockorders variable is ever used here. Can it be deleted?